### DISCUSSION TITLE
Make giggsey/libphonenumber-for-php an optional dependency?

### DISCUSSION THREAD

**Discussion by acasademont** [kreait/firebase-php#577] [Ideas]
Hi!

We're evaluating our container images sizes and we discovered a pretty big 21MB package `giggsey/libphonenumber-for-php` that's a dependency of this one, it's our biggest dependency. I inspected the code and seems like phone numbers are only used occasionally on some parts of the library. I was wondering if this dependency could be made optional with an exception on the `PhoneNumber` class asking you to composer require it if you want to use those features. What do you think? It would save us quite a lot of space in our containers.

I can work on a PR if needed, thanks!

**Comment by acasademont** [comment]
The exception would be placed here:

https://github.com/kreait/firebase-php/blob/5.x/src/Firebase/Value/PhoneNumber.php#L22

**Comment by jeromegamez** [comment]
I totally agree with you, I've been planning to remove that dependency for a long time now and just can't decide _when_ to remove it. The reason why it's included at all is to prevent a round trip to the Firebase API in case of an invalid phone number and, at the same time, to not re-invent the wheel.

But I think now seems to be as good a time as any to finally do it 😅. I'll spin something up and let you know once I have a PR ready.

**Reply by acasademont** [reply]
Lovely, thanks! :D

**Reply by jeromegamez** [reply]
@acasademont Could you have a look at #578 if it makes sense to you that way? 🙏🏼

**Comment by jeromegamez** [comment]
It took a little longer than anticipated, but release [5.17](https://github.com/kreait/firebase-php/releases/tag/5.17.0) makes `giggsey/libphonenumber-for-php` still supported, but optional, reducing the default size of the SDK with its dependencies from 27MB to less then 6MB 🥳. Thanks for bringing this up, @acasademont, I hope your deployments get a little bit faster now 💪

**Reply by acasademont** [reply]
It definitely did, these are the Docker image sizes after and before 🎉

<img width="274" alt="Screenshot 2021-03-22 at 15 33 27" src="https://user-images.githubusercontent.com/825021/112006567-1e65be80-8b24-11eb-9344-91d70f7d0934.png">
